File: Ruby/digit_digit.rb
def square_digits(num)
  arr = num.to_s.split(//)
  res = arr.map { |c| c.to_i }
  squared = res.map { |i| i * i }
  stringify = squared.map { |i| i.to_s }
  stringify.join().to_i
end

p square_digits(9119)

File: Python/digit_digit.py

# SOLUTION ONE
# def square_digits(num):
#     arr = [*str(num)]
#     res = []
#     for s in arr:
#         i = int(s)
#         res.append(str(i * i))
#     return int("".join(res))

# SOLUTION TWO
def square_digits(num):
    res = ""
    for n in str(num):
        res += str(int(n)**2)
    return int(res)


square_digits(9119)
